src/odemis/acq/stitching/__init__.py

```diff
@@ -16,6 +16,171 @@
+    """
+    tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles to compute the registration. 
+    If it's tuples, the first tile of each tuple is the “main tile”, and the following ones are 
+    dependent tiles.
+    method (REGISTER_*): REGISTER_SHIFT → ShiftRegistrar, REGISTER_IDENTITY → IdentityRegistrar
+    returns:
+        tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles as passed, but with updated 
+        MD_POS metadata
+    """
+
+    if method == REGISTER_SHIFT:
+        registrar = ShiftRegistrar()
+    elif method == REGISTER_IDENTITY:
+        registrar = IdentityRegistrar()
+    else:
+        raise ValueError("Invalid registrar %s" % (method,))
+
+    # Register tiles
+    updatedTiles = []
+    for ts in tiles:
+        # Separate tile and dependent_tiles
+        if isinstance(ts, tuple):
+            tile = ts[0]
+            dep_tiles = ts[1:]
+        else:
+            tile = ts
+            dep_tiles = None
+
+        registrar.addTile(tile, dep_tiles)
+
+    # Update positions
+    for i in range(len(tiles)):
+        # Return tuple of positions if dependent tiles are present
+        if isinstance(tiles[i], tuple):
+            tile = tiles[i][0]
+            dep_tiles = tiles[i][1:]
+
+            # Update main tile
+            md = copy.deepcopy(tile.metadata)
+            md[model.MD_POS] = registrar.getPositions()[0][i]
+            tileUpd = model.DataArray(tile, md)
+
+            # Update dependent tiles
+            tilesNew = [tileUpd]
+            for j in range(len(dep_tiles)):
+                md = copy.deepcopy(dep_tiles[j][i].metadata)
+                md[model.MD_POS] = registrar.getPositions()[1][j][i]
+                tilesNew.append(model.DataArray(tile, md))
+            tileUpd = tuple(tilesNew)
+
+        else:
+            tile = tiles[i]
+            dep_tiles = None
+
+            md = copy.deepcopy(tile.metadata)
+            md[model.MD_POS] = registrar.getPositions()[0][i]
+            tileUpd = model.DataArray(tile, md)
+
+        updatedTiles.append(tileUpd)
+
+    return updatedTiles
+
+
+def weave(tiles, method=WEAVER_MEAN):
+    """
+    tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles to compute the registration. 
+    If it's tuples, the first tile of each tuple is the “main tile”, and the following ones are dependent tiles.
+    method (WEAVER_*): WEAVER_MEAN → MeanWeaver, WEAVER_COLLAGE → CollageWeaver
+    return:
+        tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles as passed, but with updated MD_POS metadata
+    """
+
+    if method == WEAVER_MEAN:
+        weaver = MeanWeaver()
+    elif method == WEAVER_COLLAGE:
+        weaver = CollageWeaver()
+
+    for t in tiles:
+        weaver.addTile(t)
+    stitched_image = weaver.getFullImage()
+
+    return stitched_image
+
+
+def decompose_image(img, overlap=0.1, numTiles=5, method="horizontalLines", shift=True):
```

Comment:
No, don't add this function to the main code of Odemis. It's not part of the spec, and not tested by itself. It's just for the test cases, so move to stitching_test.py (or odemis.util.test, if you really want it in Odemis code)